File: tutorial_env/LC_Problems/lc_longest_substring.py

def lengthOfLongestSubstring(s: str) -> int:
    longest = 0
    # counter = 0
    left = 0
    right = 0
    set = []
    while right<(len(s)):
        if s[right] in set:
            left+=1
            set = set[(set.index(s[right])+1):]
            # set.remove(s[right])
        else:
            set.append(s[right])
            right+=1
        if (len(set)>longest):longest=len(set)
        print(set)
        # print(longest)

    return longest

s = "pwwkew"
print(lengthOfLongestSubstring(s))

File: tut_C++/lc_longest_substring.cpp
#include <cstddef>
#include <iostream>
#include <string>
// #include <vector>
#include <set>

using namespace std;

int lengthOfLongestSubstring(string s) {
    set<char> seen;
    int left = 0, right = 0;
    int longest = 0;

    while (right < s.size()) {
        if (seen.find(s[right]) == seen.end()) {
            seen.insert(s[right]);
            longest = std::max(longest, static_cast<int>(seen.size()));
            right++;
        }
        else {
            seen.erase(s[left]);
            left++;
        }
        for (auto it = seen.begin(); it != seen.end(); ++it) {
            cout << *it << ' ';
        }
        cout << endl;
    }
    return longest;
}


int main() {
    string s="pwwekk";
    cout << endl << lengthOfLongestSubstring(s) << endl;
}